File: web/models/employee.ts
export type Employee = {
  id: number
  name: string
  nickname: string | null
  job_number: string | null
  username: string
  email: string | null
  phone: string | null
  avatar: string | null
  roles: string[]
  role_ids?: number[]
  role_assignments?: EmployeeRoleAssignment[]
  is_active: boolean
  max_concurrent: number
  default_language: string
  is_super_admin: boolean
  group_ids?: number[]
  created_at: string
  updated_at: string
}

export type EmployeeRoleAssignment = {
  id: number
  key: string | null
  name: string
  is_system: boolean
  is_active: boolean
  permissions: string[]
}

export type CreateEmployeePayload = {
  name: string
  nickname?: string | null
  job_number?: string | null
  username: string
  email: string
  phone?: string | null
  password: string
  avatar?: string | null
  roles?: string[]
  role_ids?: number[]
  max_concurrent?: number
  default_language?: string
  group_ids?: number[]
}

export type UpdateEmployeePayload = {
  name?: string
  nickname?: string | null
  job_number?: string | null
  username?: string
  email?: string
  phone?: string | null
  password?: string
  avatar?: string | null
  roles?: string[]
  role_ids?: number[]
  max_concurrent?: number
  default_language?: string
  group_ids?: number[]
}

export type EmployeeStatusPayload = {
  is_active: boolean
}

File: server/app/models/employee.py
"""
Employee model — help desk employees / admins
"""
from datetime import datetime

from sqlalchemy import String, Boolean, Integer, DateTime, ForeignKey, UniqueConstraint, Index, JSON
from sqlalchemy.orm import Mapped, mapped_column, relationship

from app.db.session import Base
from app.models.base import TimestampMixin


class Employee(Base, TimestampMixin):
    __tablename__ = "employees"
    __table_args__ = (
        UniqueConstraint("tenant_id", "username", name="uq_employees_tenant_username"),
        Index("ix_employees_tenant_email", "tenant_id", "email"),
        Index("ix_employees_tenant_job_number", "tenant_id", "job_number"),
    )

    id: Mapped[int] = mapped_column(primary_key=True, autoincrement=True)
    tenant_id: Mapped[int] = mapped_column(Integer, ForeignKey("tenants.id", ondelete="CASCADE"), nullable=False)
    username: Mapped[str] = mapped_column(String(32), nullable=False)
    email: Mapped[str | None] = mapped_column(String(128), nullable=True)
    password_hash: Mapped[str] = mapped_column(String(255), nullable=False)
    display_name: Mapped[str | None] = mapped_column(String(64), nullable=True)
    roles: Mapped[list] = mapped_column(JSON, nullable=False, server_default='["agent"]')
    is_active: Mapped[bool] = mapped_column(Boolean, nullable=False, server_default="true")
    last_login_at: Mapped[datetime | None] = mapped_column(DateTime(timezone=True), nullable=True)

    name: Mapped[str] = mapped_column(String(64), nullable=False, server_default="")
    nickname: Mapped[str | None] = mapped_column(String(64), nullable=True)
    job_number: Mapped[str | None] = mapped_column(String(32), nullable=True)
    phone: Mapped[str | None] = mapped_column(String(32), nullable=True)
    avatar: Mapped[str | None] = mapped_column(String(512), nullable=True)
    max_concurrent: Mapped[int] = mapped_column(Integer, nullable=False, server_default="10")
    default_language: Mapped[str] = mapped_column(String(10), nullable=False, server_default="system")
    is_super_admin: Mapped[bool] = mapped_column(Boolean, nullable=False, server_default="false")

    tenant: Mapped["Tenant"] = relationship("Tenant", back_populates="employees")
